### ISSUE TITLE
Icon Skins and the Home.xml

### ISSUE THREAD

**Issue by SendoX** [boxeehacks/boxeehack#392]
english by google translate. LOL
Can you update the next time change the path of the icons?
in:
../720p/resources/icons/icon-home-shows.png
../720p/resources/icons/icon-home-movies.png
../720p/resources/icons/icon-home-apps.png
etc

we skin the boxee box with icon, background, etc.
After each update, we must rewrite the HOME.xml.

![al4timk8wn6i](https://f.cloud.github.com/assets/3018141/86981/d6f2dca6-64b0-11e2-9531-5a19b2b6c9d1.jpg)

It would be easier if all the icons in the folder "ICONS"

Deutsch:
Ich wollte damit sagen, das wir nicht nur den Hintergrund ändern, sondern auch die Icons.
Müssen aber nach jedem Update die Home.xml wieder umschreiben.
Es wäre einfacher zu skinnen, wenn die Icons auch im Order icons wären. so müsste man nur die png dateien aus tauschen nach belieben. So kann auch der kleine mann selber skinnen.

**Comment by coroners** [comment]
I was trying to figure out how to change the icons and customize the looks of the skin more myself. A folder for icons that could easilybe mmodified would be so kickass. SendoX if you could share your work with the skin for reference I would be really happy. More features for customizing I am all for it.  Would be glad to help out with any graphics. Love UI making.  Take a look at x-spirit.deviantart.com

**Comment by SendoX** [comment]
http://www.boxee-forum.de/board34-boxee-box/board95-boxeebox-hacks-custom-firmwares/3100-boxee-box-skinpacks-v2/

**Comment by coroners** [comment]
Thanks a lot! Going to make my own edit as soon as I get some free time.
This weekend most likely. Hope you don't mind that I use your files! PEACE!
Keep up the good work! Customize or die.

**Comment by SendoX** [comment]
here is a collection of all the changes:
http://www.boxee-forum.de/board34-boxee-box/board95-boxeebox-hacks-custom-firmwares/3136-script-sammlungen/

maybe take the makers of boxee+ a few things.

**Comment by coroners** [comment]
I made a little test to check out how it worked on my box. Take a look! http://www.mjukt.net/wp-content/uploads/2013/01/purebox.zip 

Quick question! How did you go about taking a screenshots on the boxee? Want to add a preview when I finish some skin later on. I love this shit! Thanks again for the enlightenment! I would love to give you a sion kölsch for helping out!

**Comment by Toliver182** [comment]
This is the safest way to update themes/skins 

I've i have time I will fork the git, make the changes and submit a pull request.

Then all we need to do is move the icons about with a python script to theme it.

**Comment by Toliver182** [comment]
I have been investigating this. 
The is a much better alternative to editing the .xml files and placing additional images in the icons directory.

boxee uses a texture file in which many images can be stored.

i'm making some changes that will enable you to create a texture file easily and swap it out quickly too.

this means a theme can be contained within one file and no additional changes will be needed.

i'm just going through testing now then i will upload it to my github and submit a pull request.

**Comment by coroners** [comment]
Sounds really good. Did anyone have any tips on how to take a screenshot or is theressome forum where I can ask maybe. Can't find any official forum that's why I take these questions here. Sorry.

**Comment by SendoX** [comment]
screenshots I do with my cell phone. I changeable or the pc version.
I'm sorry. my english is not so good. what is:"pull request"?

**Comment by Toliver182** [comment]
github it a place where source code can be uploaded. this is the github for boxee+. 

You can make a copy if the code. make changes then have the changes merged with the main source code.

which will go into the next release.

When i have tested this way of theming i will ask for it to be merged ( this is a pull request)

**Comment by SendoX** [comment]
thank you

**Comment by Toliver182** [comment]
I've almost completed my menu implimentation for boxee themes. with a download option so you can download them directly on boxer

**Comment by SendoX** [comment]
nice. where can I download the script. I would like to test it.

**Comment by Toliver182** [comment]
I'm going to upload it tonight. 

there is not as much freedom as some of the skins that you have been making. 

this method is safe, it will not interfere with boxee+, so if boxee+ is updated with new features the theme should continue to work.

here is a video :
http://youtu.be/BY1QduaqHLw

**Comment by Toliver182** [comment]
Here us the Theming guide http://db.tt/Rznfxec8
use that to make your themes. I'll be making a mac osx texture packer soon.

**Comment by SendoX** [comment]
great !!!!!

**Comment by SendoX** [comment]
that looks good. But where does that go? my english is too bad. can you tell me which script where I need to copy to my boxee box?
and later where the themes are stored?

**Comment by SendoX** [comment]
I understand that correctly? the function comes with the next update?

**Comment by Toliver182** [comment]
Download the theming guide above. and make your theme

when i have finsihed everything with the code for the theming then i'll submit it to @boxeehacks to join it with main boxee for the next release.

However if you want the test version you can get it here:
https://github.com/Toliver182/boxeehack

https://github.com/Toliver182/boxeehack/archive/master.zip

download the zip and replace your 'hack' folder on boxee with the one in this zip.

**Comment by SendoX** [comment]
ok. I'm on it. Also your skin purbox. But where is the homescreen background ? i have a black background

**Comment by SendoX** [comment]
my PC tells me that I lack the file "MSVCP100D.dll" for the TexturePacker

**Comment by Toliver182** [comment]
turns out you need http://www.microsoft.com/en-us/download/details.aspx?id=5555

PureBox is not my skin..
There is not an option for homescreen background. you can only have whats available on boxee already.
in this case "media/background/background.jpg"

I'm going to see if i can impliment further skinning options soon.

**Comment by SendoX** [comment]
ok. now I'm overwhelmed. lol. there may be a video tutorial on youtube? I can not find anything about it. how to use the TexturePacker. always when I start it briefly lights up a black window and I can not read what it says. what is meant by: "EC
TexturePacker.exe-media input / output Textures.xbt-nocompress "
??
screenshot:
![screen](https://f.cloud.github.com/assets/3018141/108909/d00f13a8-6a61-11e2-9e8a-ed921136dc90.png)

**Comment by Toliver182** [comment]
you need to put you media folder location in the -input and for -output where you want your Textures.xbt

i have copied the media folder to the same directory as TexturePacker.exe
![demo1](https://f.cloud.github.com/assets/3096269/109108/4142fdb6-6a6a-11e2-9867-f25d97960ba9.JPG)

This is how it should look
![demo2](https://f.cloud.github.com/assets/3096269/109107/40ef817c-6a6a-11e2-8065-cdf18e040132.JPG)

This is how it should look when completed
![demo3](https://f.cloud.github.com/assets/3096269/109113/9add423c-6a6a-11e2-9441-2e41e5bc5b96.JPG)

you can ignore the warnings. there is nothing wrong.

**Comment by SendoX** [comment]
ok..works.ask: can I copy them directly via ftp over? or it must be via a zip file at url?

**Comment by Toliver182** [comment]
You can copy it via FTP. however make sure that you copy skin.sh and have made you changes in skin.sh with your theme name.

![demo4](https://f.cloud.github.com/assets/3096269/109153/11ecad8e-6a6d-11e2-9c78-6d297aff2ef2.JPG)

splash.sh and splash are optional

**Comment by Toliver182** [comment]
so it looks like this on your boxee box
![demo 5](https://f.cloud.github.com/assets/3096269/109155/460ba46c-6a6d-11e2-9811-12ee257154d8.JPG)

**Comment by SendoX** [comment]
ok. I have a small problem.
after the installation of your hack. My home screen is always black. Black background with boxee theme. black background with purebox theme.
Why is that?
Besides others, I've found a writing error. it is twice as theme download. there would have to delete theme

def delete_theme():
    get_themes() #Load varaiables with theme names based on the directory structure

def delete_theme():
    delete_theme() #Delete theme

**Comment by Toliver182** [comment]
Yes i have already changed that, it does however do what it is supposed to. 

I did say its a test version. that issue does not occur on my boxee.

It may be best if you wait until it makes it into the main boxee+

**Comment by Toliver182** [comment]
I've submitted my request:
https://github.com/boxeehacks/boxeehack/pull/422

**Comment by SendoX** [comment]
ok. now everything works. only the home screen remains black.
Even after the improvement the boxee script runs faster.

**Comment by Toliver182** [comment]
It could be because you have previously modified your home.xml can you check your home.xml to see if it is the original?

**Comment by SendoX** [comment]
yes it is. I have uninstalled the hack. and then filled with new. the homescreen is black

**Comment by Toliver182** [comment]
how did you transfer the files to your boxee? try with this small theme http://toliverinc.co.uk/PureBox.zip

it is case sensitive.

**Comment by SendoX** [comment]
I have the skin there loaded down
http://toliverinc.co.uk/PureBox.zip
without success. the home screen remains black.
only in fanart mode it still works.
here is the link to my home.xml
http://www.sendox.de/boxee/Home.zip
because you can check it

**Comment by Toliver182** [comment]
Ok i just reset by boxee to factory settings. then upgraded to the latest firmware then installed boxee+ again using my repo and it works perfectly.

if you want to try it you can follow the instructions here to uninstall
    If you want to completely uninstall you can run the uninstaller on your USB stick. Just go into Settings -> Network -> Servers, and change your hostname into "boxeebox; sh /media/BOXEE/uninstall.sh".

Then if you want to install from my repo 
download the boxee+ from here http://boxeeplus.com/
follow the instructions. 

but edit install.sh by changing this line in install.sh
/opt/local/bin/curl -L https://github.com/boxeehacks/boxeehack/archive/master.zip -o boxeehack.zip
to this:
/opt/local/bin/curl -L https://github.com/Toliver182/boxeehack/archive/master.zip -o boxeehack.zip

then follow the rest of the install instructions.

or you can wait to see if it gets merged into the next release.

**Comment by SendoX** [comment]
i wait for the next release
